pkg/distributor/distributor.go

```diff
@@ -356,7 +360,8 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		receivedSamples: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
 			Name: "cortex_distributor_received_samples_total",
 			Help: "The total number of received samples, excluding rejected and deduped samples.",
-		}, []string{"user"}),
+		}, []string{"user", "attrib"}),
```

Comment:
As mentioned offline: can we use `group` just like in the existing feature for push errors tracking?

Reply:
my issue is what if a user configured using group and cost attribution feature together? I should have also touching the same metrics `cortex_discarded_samples_total`, i think group can be miss leading when activate 2 features together.

Reply:
> my issue is what if a user configured using group and cost attribution feature together?

Do we have any customer who wants to configure these two (for me one) features with different labels? I feel like the _active groups_ thing is the predecessor feature to _attribution_.

Reply:
I mostly agree with you, just active groups is used for tracking discarded samples with reasons, for the cost breakdown, we don't need reason label, and in order to have it, we will need to store a detailed counter per reason. I start thinking that have a completely new metrics exported directly to a new registry making more sense for the following reasons, I will list this in design doc:
- lower cardinality: the new exported metrics would have only userID and cost-attribution-label as labels, this should be much easier, and no breaking changes for active group feature, less pressure on current registry
- can have high scraping interval, more flexible
- having dynamic label that corresponding to user config instead of hard coded "group"
- align with how loki/tempo's way of approaching this